#  Business Sale Ana
In this notebook we will make some analysis using SQL to answer get some vision about our business our database called " chinook.db " ,let's start by connect our data base chinook.db

In [1]:
%%capture
%load_ext sql
%sql sqlite:///chinook.db

'Connected: None@chinook.db'

 exploring tables & views on out database 

In [3]:
%%sql
SELECT
    name,
    type
FROM sqlite_master
WHERE type IN ("table","view");

Done.


[('album', 'table'),
 ('artist', 'table'),
 ('customer', 'table'),
 ('employee', 'table'),
 ('genre', 'table'),
 ('invoice', 'table'),
 ('invoice_line', 'table'),
 ('media_type', 'table'),
 ('playlist', 'table'),
 ('playlist_track', 'table'),
 ('track', 'table')]

from above we find that our data have 11 table only & there are no view, one can look into the table and see the tuples and attributes.

In [4]:
%%sql
select *from album;

Done.


[(1, 'For Those About To Rock We Salute You', 1),
 (2, 'Balls to the Wall', 2),
 (3, 'Restless and Wild', 2),
 (4, 'Let There Be Rock', 1),
 (5, 'Big Ones', 3),
 (6, 'Jagged Little Pill', 4),
 (7, 'Facelift', 5),
 (8, 'Warner 25 Anos', 6),
 (9, 'Plays Metallica By Four Cellos', 7),
 (10, 'Audioslave', 8),
 (11, 'Out Of Exile', 8),
 (12, 'BackBeat Soundtrack', 9),
 (13, 'The Best Of Billy Cobham', 10),
 (14, 'Alcohol Fueled Brewtality Live! [Disc 1]', 11),
 (15, 'Alcohol Fueled Brewtality Live! [Disc 2]', 11),
 (16, 'Black Sabbath', 12),
 (17, 'Black Sabbath Vol. 4 (Remaster)', 12),
 (18, 'Body Count', 13),
 (19, 'Chemical Wedding', 14),
 (20, 'The Best Of Buddy Guy - The Millenium Collection', 15),
 (21, 'Prenda Minha', 16),
 (22, 'Sozinho Remix Ao Vivo', 16),
 (23, 'Minha Historia', 17),
 (24, 'Afrociberdelia', 18),
 (25, 'Da Lama Ao Caos', 18),
 (26, 'Acústico MTV [live]', 19),
 (27, 'Cidade Negra - Hits', 19),
 (28, 'Na Pista', 20),
 (29, 'Axé Bahia 2001', 21),
 (30, 'BBC Sessions [Disc 1] [live]', 22),
 (31, 'Bongo Fury', 23),
 (32, 'Carnaval 2001', 21),
 (33, 'Chill: Brazil (Disc 1)', 24),
 (34, 'Chill: Brazil (Disc 2)', 6),
 (35, 'Garage Inc. (Disc 1)', 50),
 (36, 'Greatest Hits II', 51),
 (37, 'Greatest Kiss', 52),
 (38, 'Heart of the Night', 53),
 (39, 'International Superhits', 54),
 (40, 'Into The Light', 55),
 (41, 'Meus Momentos', 56),
 (42, 'Minha História', 57),
 (43, 'MK III The Final Concerts [Disc 1]', 58),
 (44, 'Physical Graffiti [Disc 1]', 22),
 (45, 'Sambas De Enredo 2001', 21),
 (46, 'Supernatural', 59),
 (47, 'The Best of Ed Motta', 37),
 (48, 'The Essential Miles Davis [Disc 1]', 68),
 (49, 'The Essential Miles Davis [Disc 2]', 68),
 (50, 'The Final Concerts (Disc 2)', 58),
 (51, "Up An' Atom", 69),
 (52, 'Vinícius De Moraes - Sem Limite', 70),
 (53, 'Vozes do MPB', 21),
 (54, 'Chronicle, Vol. 1', 76),
 (55, 'Chronicle, Vol. 2', 76),
 (56, 'Cássia Eller - Coleção Sem Limite [Disc 2]', 77),
 (57, 'Cássia Eller - Sem Limite [Disc 1]', 77),
 (58, 'Come Taste The Band', 58),
 (59, 'Deep Purple In Rock', 58),
 (60, 'Fireball', 58),
 (61, "Knocking at Your Back Door: The Best Of Deep Purple in the 80's", 58),
 (62, 'Machine Head', 58),
 (63, 'Purpendicular', 58),
 (64, 'Slaves And Masters', 58),
 (65, 'Stormbringer', 58),
 (66, 'The Battle Rages On', 58),
 (67, "Vault: Def Leppard's Greatest Hits", 78),
 (68, 'Outbreak', 79),
 (69, 'Djavan Ao Vivo - Vol. 02', 80),
 (70, 'Djavan Ao Vivo - Vol. 1', 80),
 (71, 'Elis Regina-Minha História', 41),
 (72, 'The Cream Of Clapton', 81),
 (73, 'Unplugged', 81),
 (74, 'Album Of The Year', 82),
 (75, 'Angel Dust', 82),
 (76, 'King For A Day Fool For A Lifetime', 82),
 (77, 'The Real Thing', 82),
 (78, 'Deixa Entrar', 83),
 (79, 'In Your Honor [Disc 1]', 84),
 (80, 'In Your Honor [Disc 2]', 84),
 (81, 'One By One', 84),
 (82, 'The Colour And The Shape', 84),
 (83, 'My Way: The Best Of Frank Sinatra [Disc 1]', 85),
 (84, 'Roda De Funk', 86),
 (85, 'As Canções de Eu Tu Eles', 27),
 (86, 'Quanta Gente Veio Ver (Live)', 27),
 (87, 'Quanta Gente Veio ver--Bônus De Carnaval', 27),
 (88, 'Faceless', 87),
 (89, 'American Idiot', 54),
 (90, 'Appetite for Destruction', 88),
 (91, 'Use Your Illusion I', 88),
 (92, 'Use Your Illusion II', 88),
 (93, 'Blue Moods', 89),
 (94, 'A Matter of Life and Death', 90),
 (95, 'A Real Dead One', 90),
 (96, 'A Real Live One', 90),
 (97, 'Brave New World', 90),
 (98, 'Dance Of Death', 90),
 (99, 'Fear Of The Dark', 90),
 (100, 'Iron Maiden', 90),
 (101, 'Killers', 90),
 (102, 'Live After Death', 90),
 (103, 'Live At Donington 1992 (Disc 1)', 90),
 (104, 'Live At Donington 1992 (Disc 2)', 90),
 (105, 'No Prayer For The Dying', 90),
 (106, 'Piece Of Mind', 90),
 (107, 'Powerslave', 90),
 (108, 'Rock In Rio [cd1]', 90),
 (109, 'Rock In Rio [cd2]', 90),
 (110, 'Seventh Son of a Seventh Son', 90),
 (111, 'Somewhere in Time', 90),
 (112, 'The Number of The Beast', 90),
 (113, 'The X Factor', 90),
 (114, 'Virtual XI', 90),
 (115, 'Sex Machine', 91),
 (116, 

In [22]:
%%sql 
with us as 
(select il.* from invoice_line il inner join invoice inv on inv.invoice_id=il.invoice_id inner join customer c on c.customer_id=inv.customer_id where c.country="USA")

select g.name , count(us.invoice_line_id) track_sold, CAST(count(us.invoice_line_id)as float)/(select count(*) from us) per_sold from us
inner join track t on t.track_id=us.track_id 
inner join genre g on g.genre_id=t.genre_id
group by 1
order by 2 desc 

Done.


[('Rock', 561, 0.5337773549000951),
 ('Alternative & Punk', 130, 0.12369172216936251),
 ('Metal', 124, 0.11798287345385347),
 ('R&B/Soul', 53, 0.05042816365366318),
 ('Blues', 36, 0.03425309229305423),
 ('Alternative', 35, 0.03330161750713606),
 ('Latin', 22, 0.02093244529019981),
 ('Pop', 22, 0.02093244529019981),
 ('Hip Hop/Rap', 20, 0.019029495718363463),
 ('Jazz', 14, 0.013320647002854425),
 ('Easy Listening', 13, 0.012369172216936251),
 ('Reggae', 6, 0.005708848715509039),
 ('Electronica/Dance', 5, 0.004757373929590866),
 ('Classical', 4, 0.003805899143672693),
 ('Heavy Metal', 3, 0.0028544243577545195),
 ('Soundtrack', 2, 0.0019029495718363464),
 ('TV Shows', 1, 0.0009514747859181732)]


Based on the sales of tracks across different genres in the USA, we should purchase the new albums by the following artists:

* Red Tone (Punk)
* Slim Jim Bites (Blues)
* Meteor and the Girls (Pop)
It's worth keeping in mind that combined, these three genres only make up only 17% of total sales, so we should be on the lookout for artists and albums from the 'rock' genre, which accounts for 53% of sales.

In [25]:
%%sql 
with sale as 
(select c.customer_id,c.support_rep_id,sum(inv.total) total from customer c 
inner join invoice inv on inv.customer_id=c.customer_id group by 1,2)

select e.first_name||" "||e.last_name name,
       e.hire_date,
        sum(s.total) total_sale
        from sale s 
        inner join employee e on e.employee_id=s.support_rep_id group by 1

Done.


[('Jane Peacock', '2017-04-01 00:00:00', 1731.5099999999998),
 ('Margaret Park', '2017-05-03 00:00:00', 1584.0000000000002),
 ('Steve Johnson', '2017-10-17 00:00:00', 1393.92)]


While there is a 20% difference in sales between Jane (the top employee) and Steve (the bottom employee), the difference roughly corresponds with the differences in their hiring dates.

In [29]:
%%sql 
with coun as 
(select 
  case 
    when(select count(*)from customer where country=c.country )=1 THEN "Other"
    else c.country 
    end as country ,c.customer_id,il.* from invoice_line il 
    inner join invoice i on i.invoice_id=il.invoice_id
    inner join customer c on c.customer_id=i.customer_id
)
select
 country,
    customers,
    total_sales,
    average_order,
    customer_lifetime_value
from (select
     country,
      count(distinct customer_id) customers,
      sum(unit_price) total_sales,
      SUM(unit_price) / count(distinct customer_id) customer_lifetime_value,
        SUM(unit_price) / count(distinct invoice_id) average_order,
      CASE
       when country="Other"  then 1
      else 0
      end  as sort 
      from coun
      group by country 
      order by sort asc, total_sales desc
     )

Done.


[('USA', 13, 1040.490000000008, 7.942671755725252, 80.03769230769292),
 ('Canada', 8, 535.5900000000034, 7.047236842105309, 66.94875000000043),
 ('Brazil', 5, 427.68000000000245, 7.011147540983647, 85.53600000000048),
 ('France', 5, 389.0700000000021, 7.781400000000042, 77.81400000000042),
 ('Germany', 4, 334.6200000000016, 8.161463414634186, 83.6550000000004),
 ('Czech Republic', 2, 273.24000000000103, 9.108000000000034, 136.62000000000052),
 ('United Kingdom', 3, 245.52000000000078, 8.768571428571457, 81.84000000000026),
 ('Portugal', 2, 185.13000000000022, 6.3837931034482835, 92.56500000000011),
 ('India', 2, 183.1500000000002, 8.72142857142858, 91.5750000000001),
 ('Other', 15, 1094.9400000000085, 7.448571428571486, 72.99600000000056)]

We conclude:
* USA have the maximum of our sales with total sales 1040USD taking about 22% of our sales , but we have about 13 customers so our sales average per customer about 80USD with sales average per order 7.94USD.
* Czech Republic have the Maximum sales average per customer 136.62USD and by order 9.11$ although the it take only 6% of our sales , so we can increase our sales on it by increasing number of customers,also on United Kingdom, Portugal & India we have an opportunity to increasing our customers by increasing our sales there as sales average per customer & average order also is better than USA except average order in portugal.